File: src/main/java/dev/slne/surf/lobby/jar/command/subcommand/setting/ParkourSettingAreaCommand.java
package dev.slne.surf.lobby.jar.command.subcommand.setting;

import com.sk89q.worldedit.IncompleteRegionException;
import com.sk89q.worldedit.LocalSession;
import com.sk89q.worldedit.WorldEdit;
import com.sk89q.worldedit.bukkit.WorldEditPlugin;
import com.sk89q.worldedit.math.BlockVector3;
import com.sk89q.worldedit.regions.Region;
import dev.jorel.commandapi.CommandAPICommand;
import dev.slne.surf.lobby.jar.JumpAndRun;
import dev.slne.surf.lobby.jar.PluginInstance;
import net.kyori.adventure.text.Component;
import org.bukkit.Location;

public class ParkourSettingAreaCommand extends CommandAPICommand {

  public ParkourSettingAreaCommand(String commandName) {
    super(commandName);

    withPermission("jumpandrun.command.setting.area");

    executesPlayer((player, args) -> {

      if(!PluginInstance.instance().worldedit()) {
        player.sendMessage(PluginInstance.prefix().append(Component.text("Bitte installiere WorldEdit um diesen Command auszuführen.")));
        return;
      }

      WorldEditPlugin worldEditPlugin = PluginInstance.instance().worldEditInstance();

      if (worldEditPlugin == null) {
        player.sendMessage(PluginInstance.prefix().append(Component.text("WorldEdit ist nicht verfügbar!")));
        return;
      }

      LocalSession session = worldEditPlugin.getSession(player);
      try {
        Region region = session.getSelection(session.getSelectionWorld());


        BlockVector3 pos1 = region.getMinimumPoint();
        BlockVector3 pos2 = region.getMaximumPoint();

        JumpAndRun jumpAndRun = PluginInstance.instance().jumpAndRunProvider().jumpAndRun();
        jumpAndRun.setPosOne(new Location(player.getWorld(), pos1.getX(), pos1.getY(), pos1.getZ()));
        jumpAndRun.setPosTwo(new Location(player.getWorld(), pos2.getX(), pos2.getY(), pos2.getZ()));

        player.sendMessage(PluginInstance.prefix().append(Component.text("Du hast die Arena erfolgreich neu definiert.")));

      } catch (IncompleteRegionException e) {
        player.sendMessage(PluginInstance.prefix().append(Component.text("Bitte wähle einen Bereich mit WorldEdit aus!")));
      }
    });
  }
}

File: src/main/java/dev/slne/surf/lobby/jar/command/subcommand/setting/ParkourSettingAreaCommand.kt
package dev.slne.surf.lobby.jar.command.subcommand.setting

import dev.jorel.commandapi.CommandAPICommand
import dev.jorel.commandapi.arguments.LocationArgument
import dev.jorel.commandapi.arguments.WorldArgument
import dev.jorel.commandapi.executors.CommandArguments
import dev.jorel.commandapi.executors.PlayerCommandExecutor
import dev.slne.surf.lobby.jar.service.JumpAndRunService
import dev.slne.surf.lobby.jar.PluginInstance
import net.kyori.adventure.text.Component
import org.bukkit.Location
import org.bukkit.entity.Player
import org.bukkit.util.Vector

class ParkourSettingAreaCommand(commandName: String) : CommandAPICommand(commandName) {
    init {
        withPermission("jumpandrun.command.setting.area")
        withArguments(LocationArgument("pos1"), LocationArgument("pos2"))
        withOptionalArguments(WorldArgument("world"))

        executesPlayer(PlayerCommandExecutor { player: Player, args: CommandArguments ->
            val pos1: Location = args.getUnchecked("pos1") ?: return@PlayerCommandExecutor
            val pos2: Location = args.getUnchecked("pos2") ?: return@PlayerCommandExecutor

            val max = Vector(pos1.x, pos1.y, pos1.z)
            val min = Vector(pos2.x, pos2.y, pos2.z)

            JumpAndRunService.jumpAndRun.boundingBox.max.setX(max.x).setY(max.y).setZ(max.z)
            JumpAndRunService.jumpAndRun.boundingBox.min.setX(min.x).setY(min.y).setZ(min.z)
            JumpAndRunService.jumpAndRun.world = args.getUnchecked("world")

            player.sendMessage(PluginInstance.prefix.append(Component.text("Du hast die Arena erfolgreich neu definiert.")))
        })
    }
}
